webapp_frontend/src/components/pages/Registration.tsx

```diff
@@ -0,0 +1,162 @@
+    const [passwordInformationLowercase, setPasswordInformationLowercase] = useState<boolean>(false);
+    const [passwordInformationUppercase, setPasswordInformationUppercase] = useState<boolean>(false);
+    const [passwordInformationNumber, setPasswordInformationNumber] = useState<boolean>(false);
+    const [passwordsMatch, setPasswordsMatch] = useState<boolean>(true);
+    const [alertMessage, setAlertMessage] = useState<string>("Error 404: No Message found.");
+    const [alertVariant, setAlertColor] = useState<"primary" | "secondary" | "success" | "danger" | "warning" | "info" | "light" | "dark">("success");
+    const [alertVisibility, setAlertVisibility] = useState<boolean>(false);
+
+    useEffect(() => {
+        reviewPasswordMatch()
+        // eslint-disable-next-line
+    },[passwordConfirmation, password])
+
+    const handleSubmit = async (event: FormEvent) => {
+        console.log("[REGISTRATION] handleSubmit")
+        event.preventDefault();
+        reviewPasswordMatch();
+        if (!username){
+            setAlertColor("danger");
+            setAlertMessage("Error: Please choose an username.")
+            handleAlertVisibility(3500)
+        } else if (!passwordsMatch) {
+            setAlertColor("danger");
+            setAlertMessage("Error: Password and password confirmation must match.")
+            handleAlertVisibility(3500)
+        } else if (!passwordInformationNumber || !passwordInformationLowercase || !passwordInformationUppercase || !passwordInformationLength){
+            setAlertColor("danger");
+            setAlertMessage("Error: Please pay attention to the notes below the input field.");
+            handleAlertVisibility(3500)
+        } else {
+            await registerNewUser(username, password, passwordConfirmation)
+                .then(res => {
+                    setAlertMessage("Worked: " + (res.outputMessage ? res.outputMessage : (res.httpStatus + " " + res.httpMessage)));
+                    setAlertColor("success");
+                    console.table(res);
+                })
+                .catch(err => {
+                    setAlertColor("danger");
+                    setAlertMessage("Error: " + (err.outputMessage ? err.outputMessage : (err.httpStatus + " " + err.httpMessage)))
+                    console.table(err)
+                })
+                .finally(() => handleAlertVisibility(3500))
+        }
+    }
+
+    const handleAlertVisibility = (duration: number) => {
+        if (!alertVisibility) {
+            setAlertVisibility(true);
+            setTimeout(() => {
+                setAlertVisibility(false)
+            }, duration)
+        }
+    }
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+        let value = event.target.value;
+        value = deleteSpaces(value);
+        setPasswordInformationLength(!notMinStrLength(value, MIN_PASSWORD_LENGTH));
```

Comment:
max lenght for the BE is 20, so this should also be here. @open-schnick but what if i want my password to be longer than 20 chars?

Reply:
done

Reply:
@qvalentin well we could implement a config key. How bout roadmap?